src/main/java/org/apache/commons/compress/archivers/zip/ZipFile.java

```diff
@@ -1081,16 +1082,26 @@ private boolean startsWithLocalFileHeader() throws IOException {
     }
 
     /**
+     * Creates new BoundedInputStream, according to implementation of
+     * underlying archive channel.
+     */
+    private BoundedInputStream createBoundedInputStream(long start, long remaining) {
+        return archive instanceof FileChannel ?
+            new BoundedFileChannelInputStream(start, remaining) :
+            new BoundedInputStream(start, remaining);
+    }
+
+    /**
      * InputStream that delegates requests to the underlying
      * SeekableByteChannel, making sure that only bytes from a certain
      * range can be read.
      */
     private class BoundedInputStream extends InputStream {
-        private static final int MAX_BUF_LEN = 8192;
-        private final ByteBuffer buffer;
-        private long remaining;
-        private long loc;
-        private boolean addDummyByte = false;
+        protected static final int MAX_BUF_LEN = 8192;
+        protected final ByteBuffer buffer;
+        protected long remaining;
+        protected long loc;
+        protected boolean addDummyByte = false;
 
```

Comment:
They can be package protected too, it's in private class anyway. The fields are accessed by the FileChannel specialization so private won't suffice.